Gem/Code/Source/Components/NetworkPlayerMovementComponent.cpp

```diff
@@ -163,20 +240,27 @@ namespace MultiplayerSample
             }
         }
 
-        // Not moving?
-        if (fwdBack == 0.0f && leftRight == 0.0f)
-        {
-            SetVelocity(AZ::Vector3::CreateZero());
-        }
-        else
+        if (fwdBack != 0.0f || leftRight != 0.0f)
         {
             const float stickInputAngle = AZ::Atan2(leftRight, fwdBack);
             const float currentHeading = GetNetworkTransformComponentController()->GetRotation().GetEulerRadians().GetZ();
-            const float targetHeading =
-                NormalizeHeading(currentHeading + stickInputAngle); // Update current heading with stick input angles
+            const float targetHeading = NormalizeHeading(currentHeading + stickInputAngle);
             const AZ::Vector3 fwd = AZ::Vector3::CreateAxisY();
-            SetVelocity(AZ::Quaternion::CreateRotationZ(targetHeading).TransformVector(fwd) * speed);
+
+            // instant acceleration for now
+            const AZ::Vector3 moveVelocity = AZ::Quaternion::CreateRotationZ(targetHeading).TransformVector(fwd) * speed;
```

Comment:
It would be great to have it be impulse based in the future.

Reply:
How do you mean? Impulses being an instantaneous force applied that frame seems like what this is doing? Or are you suggesting we have an API for adding impulses and then we apply the resulting velocity in one place?

Reply:
I'm guessing he's thinking ahead to knockback. Knockback is a complex enough feature we need to sit down and discuss how best to approach it. What's the desired visual effect? How should it affect movement and gameplay, do we want characters to ragdoll under these circumstances, etc.. Then inevitably we'll need to try a bunch of stuff and iterate on it until it actually feels and plays well.

Reply:
For now I'm thinking we can expose an API that lets you add an instant "impulse" velocity to the base velocity that will be scaled back over time based on ground/air/water deceleration - another thing I was going to add to the general velocity and not just gravity but I was just trying to get something useable up for testing the level so that is unblocked while I get back to the other input fixes.